torchbenchmark/util/extra_args.py

```diff
@@ -113,8 +116,16 @@ def apply_decoration_args(model: 'torchbenchmark.util.model.BenchmarkModel', dar
             import torch
             model.add_context(lambda: torch.cuda.amp.autocast(dtype=torch.float16), stage=TEST_STAGE.FORWARD)
     elif dargs.precision == "amp_bf16":
-        import torch
-        model.add_context(lambda: torch.cpu.amp.autocast(dtype=torch.bfloat16))
+        if model.test == "eval":
+            import torch
+            model.add_context(lambda: torch.cpu.amp.autocast(dtype=torch.bfloat16))
```

Comment:
This context only applies when the device is `cpu`, is that correct?

Reply:
Yes, the device check is in https://github.com/pytorch/benchmark/pull/2198/files#diff-6ccf656b90a64ee9ee4e55aec794320710e717b65271baeae74e69940524bb6aR35.

Reply:
Can we add an assert here? (see above comment)

Reply:
Done.